### PR TITLE
maven: 3.8.6 -> 3.9.2

### PR DESCRIPTION
**Pull request by Tillerino**
###### Description of changes

Closes #230162, closes #229144.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Tillerino** [comment]
Result of `nixpkgs-review pr 238746` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>cryptomator</li>
    <li>dbeaver</li>
    <li>digital</li>
    <li>exhibitor</li>
    <li>forge-mtg</li>
    <li>gephi</li>
    <li>global-platform-pro</li>
    <li>java-language-server</li>
    <li>jd-cli</li>
    <li>jetbrains.idea-community</li>
    <li>jetbrains.idea-ultimate</li>
    <li>kotlin-language-server</li>
    <li>maven</li>
    <li>schemaspy</li>
  </ul>
</details>

**Comment by wegank** [comment]
@ofborg build maven maven.passthru.tests

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 238746` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>dbeaver</li>
    <li>exhibitor</li>
    <li>forge-mtg</li>
    <li>global-platform-pro</li>
    <li>java-language-server</li>
    <li>jd-cli</li>
    <li>kotlin-language-server</li>
    <li>maven</li>
    <li>schemaspy</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Comment by wegank** [comment]
Thank you!

File: pkgs/applications/misc/dbeaver/default.nix [hunk 1]
```diff
@@ -37,7 +37,7 @@ in
     hash = "sha256-TUdtrhQ1JzqZx+QNauNA1P/+WDSSeOGIgGX3SdS0JTI=";
   };
 
-  mvnHash = "sha256-87pf7XRXCuZlAbL54pX+a5Lo/874DmUr/W37/V+5YpQ=";
+  mvnHash = "sha256-ERZYDsPxp1YXteSmunFIgTGZUYqjZJhqrNytLnIUNBQ=";
```
Comment (Tillerino) [inline pkgs/applications/misc/dbeaver/default.nix:40]: (note to self) This package is up-to-date and I opened a PR to enforce all maven plugin versions to be set. With the next update of this package, the hash should no longer be affected by the Maven version.

File: pkgs/development/tools/global-platform-pro/default.nix [hunk 2]
```diff
@@ -32,7 +32,7 @@ stdenv.mkDerivation rec {
     '';
     outputHashAlgo = "sha256";
     outputHashMode = "recursive";
-    outputHash = "1qwgvz6l5wia8q5824c9f3iwyapfskljhqf1z09fw6jjj1jy3b15";
+    outputHash = "sha256-rRLsCTY3fEAvGRDvNXqpjac2Gb5fdlyhK2wTK5CVN9k=";
```
Comment (Tillerino) [inline pkgs/development/tools/global-platform-pro/default.nix:35]: This was not up-to-date at all. Opened a PR about the plugin versions anyway: martinpaljak/GlobalPlatformPro#321

File: pkgs/development/tools/java/java-language-server/default.nix [hunk 3]
```diff
@@ -24,7 +24,7 @@ javaPackages.mavenfod rec {
 
   mvnFetchExtraArgs.dontConfigure = true;
   mvnParameters = "-DskipTests";
-  mvnHash = "sha256-2MVlF3QIWiDvUlnMH4RLi2Od57aoh8zK/OmHqztOnZ4=";
+  mvnHash = "sha256-bzYBSrCS9Kp+qnVO60h915Or1VWabphwLEu6lcBULuc=";
```
Comment (Tillerino) [inline pkgs/development/tools/java/java-language-server/default.nix:27]: (note to self) this project seems kind of dead. there are really trivial PRs that haven't been merged in months :(

I threw in the enforce PR just to kind of see what will happen: georgewfraser/java-language-server#259

File: pkgs/servers/exhibitor/default.nix [hunk 4]
```diff
@@ -9,7 +9,7 @@ stdenv.mkDerivation rec {
     sha256 = "07vikhkldxy51jbpy3jgva6wz75jksch6bjd6dqkagfgqd6baw45";
     rev = "5fcdb411d06e8638c2380f7acb72a8a6909739cd";
   };
-  mavenDependenciesSha256 = "00r69n9hwvrn5cbhxklx7w00sjmqvcxs7gvhbm150ggy7bc865qv";
+  mavenDependenciesSha256 = "sha256-oGd8PFF1Z9BElNsgqbC/llQm8j/T+T01VwvSw3tTU0E=";
```
Comment (Tillerino) [inline pkgs/servers/exhibitor/default.nix:12]: @wegank From https://github.com/soabase/exhibitor:

> This repository has been archived by the owner on Apr 23, 2020. It is now read-only.

That's three years ago. Is that something that would be removed eventually? There's a whole service that depends on this in NixOS, though, if I understand this correctly: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/misc/exhibitor.nix#L382

File: pkgs/tools/security/jd-cli/default.nix [hunk 5]
```diff
@@ -11,7 +11,7 @@ javaPackages.mavenfod rec {
     hash = "sha256-rRttA5H0A0c44loBzbKH7Waoted3IsOgxGCD2VM0U/Q=";
   };
 
-  mvnHash = "sha256-EIam0rxmCBs/mpMck6ePFoQBQ6KYYNqJKVE32gdXPfE=";
+  mvnHash = "sha256-kLpjMj05uC94/5vGMwMlFzLKNFOKeyNvq/vmB6pHTAo=";
```
Comment (Tillerino) [inline pkgs/tools/security/jd-cli/default.nix:14]: (note to self) This package was up to date. Opened an enforce PR: intoolswetrust/jd-cli#94

File: pkgs/applications/misc/dbeaver/default.nix (hunk 1 above)
Comment (Tillerino) [reply pkgs/applications/misc/dbeaver/default.nix:40]: Ok, I got to this repo now and opened the enforce PR: dbeaver/dbeaver#20341

I don't think it's worth waiting for a merge and release, so please disregard w.r.t. #232348

File: pkgs/servers/exhibitor/default.nix (hunk 4 above)
Comment (wegank) [reply pkgs/servers/exhibitor/default.nix:12]: Dropped.

File: pkgs/tools/security/jd-cli/default.nix (hunk 5 above)
Comment (Tillerino) [reply pkgs/tools/security/jd-cli/default.nix:14]: this was merged

File: pkgs/development/tools/java/java-language-server/default.nix (hunk 3 above)
Comment (Tillerino) [reply pkgs/development/tools/java/java-language-server/default.nix:27]: nevermind, this was merged :)

**Pull request merged**